src/nan_preserving_float.rs

```diff
@@ -96,6 +96,7 @@ macro_rules! float {
             }
         }
 
+        #[allow(clippy::cmp_owned)]
```

Comment:
When silencing clippy please leave a short comment on why you think it's fine here: is that a false positive? is it fine because of other reasons?

Reply:
I actually tried the suggestion but it fails the tests then. Clippy seems to base the suggestion on *some* runs of the macro but does not pay attention to the whole.